ethcore/src/ethereum/ethash.rs

```diff
@@ -186,24 +205,38 @@ impl Engine<EthereumMachine> for Arc<Ethash> {
 		let author = *LiveBlock::header(&*block).author();
 		let number = LiveBlock::header(&*block).number();
 
+		// Applies EIP-649 reward.
 		let reward = if number >= self.ethash_params.eip649_transition {
 			self.ethash_params.eip649_reward.unwrap_or(self.ethash_params.block_reward)
 		} else {
 			self.ethash_params.block_reward
 		};
 
+		// Applies ECIP-1017 eras.
 		let eras_rounds = self.ethash_params.ecip1017_era_rounds;
 		let (eras, reward) = ecip1017_eras_block_reward(eras_rounds, reward, number);
 
 		let n_uncles = LiveBlock::uncles(&*block).len();
 
-		// Bestow block reward
+		// Bestow block rewards.
 		let result_block_reward = reward + reward.shr(5) * U256::from(n_uncles);
 		let mut uncle_rewards = Vec::with_capacity(n_uncles);
 
-		self.machine.add_balance(block, &author, &result_block_reward)?;
+		if number >= self.ethash_params.mcip3_transition {
```

Comment:
Is it possible to write tests for that?

Reply:
Yes, would be lucky to do that. Do I just add tests at the bottom of ethash.rs? Can I reuse the `test_spec()` or should I create a new test chain that includes the transition?